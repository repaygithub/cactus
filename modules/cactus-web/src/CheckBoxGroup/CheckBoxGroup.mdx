---
name: CheckBoxGroup
menu: Components
---

import PropsTable from 'website-src/components/PropsTable'
import { livePreviewStyle } from '../helpers/constants'
import CheckBoxGroup from './CheckBoxGroup'
import cactusTheme from '@repay/cactus-theme'
import { LiveProvider, LiveEditor, LiveError, LivePreview } from 'react-live'

# CheckBoxGroup

`CheckBoxGroup` is a wrapper for a group of associated checkboxes which behaves like other field components.

## Basic Usage

`CheckBoxGroup` renders a `fieldset`, and any checkboxes in the group should be rendered and children.
`required` and `disabled` props are forwarded to all of the checkboxes, but you can still disable individual checkboxes if you need to.

`CheckBoxGroup.Item` is a wrapper for `CheckBoxField` with the `required` prop removed so that it can be left out without causing TS errors.

### Try It out

export const code = `
<CheckBoxGroup name="checkboxes" label="A CheckBox Group">
  <CheckBoxGroup.Item name="opt1" label="Option 1" />
  <CheckBoxGroup.Item name="opt2" label="Option 2" />
  <CheckBoxGroup.Item name="opt3" label="Option 3" />
</CheckBoxGroup>
`

<LiveProvider code={code} scope={{ CheckBoxGroup }}>
  <LiveEditor style={livePreviewStyle} />
  <LiveError />
  <LivePreview />
</LiveProvider>

## Best Practices

- Use meaningful labels to ensure accessibility.
- `CheckBoxGroup` props will only be forwarded to the individual checkboxes if they are direct children of `CheckBoxGroup`.
If you use an intermediate component, it will have to accept the forwarded props and pass them on.
- You can control the values using one of two methods, which should not be mixed:
  - Pass `checked` to `CheckBoxGroup`. This should be an object, mapping the checkbox names to their boolean values.
  - Pass `checked` or `defaultChecked` to `CheckBoxGroup.Item`s.
- Passing a `ref` to `CheckBoxGroup` will be set on the `fieldset` element.
- `onChange`, `onFocus` and `onBlur` handlers are not forwarded to individual checkboxes. They are associated with the `fieldset` and rely on event delegation to capture changes.

## Properties

<PropsTable of={CheckBoxGroup} />
