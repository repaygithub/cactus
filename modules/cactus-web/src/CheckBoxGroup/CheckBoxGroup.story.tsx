import { boolean, select, text } from '@storybook/addon-knobs'
import { Meta } from '@storybook/react/types-6-0'
import React, { useState } from 'react'

import { CheckBoxGroup, Flex } from '../'

export default {
  title: 'CheckBoxGroup',
  component: CheckBoxGroup,
} as Meta

export const BasicUsage = (): React.ReactElement => (
  <CheckBoxGroup
    name="checkboxes"
    id="cbg"
    label={text('label', 'My Label')}
    disabled={boolean('disabled', false)}
    onChange={(e: any) => console.log(`'${e.target.name}' changed: ${e.target.checked}`)}
    onFocus={(e) => console.log(`'${e.target.name}' focused`)}
    onBlur={(e) => console.log(`'${e.target.name}' blurred`)}
    tooltip={text('tooltip', 'Check some boxes')}
    autoTooltip={boolean('autoTooltip', true)}
    error={text('error', '')}
    success={text('success', '')}
    warning={text('warning', '')}
    disableTooltip={select('disableTooltip', [false, true, undefined], false)}
    alignTooltip={select('alignTooltip', ['left', 'right'], 'right')}
  >
    <CheckBoxGroup.Item name="option-1" label="Option 1" />
    <CheckBoxGroup.Item name="option-2" label="Option 2" />
    <CheckBoxGroup.Item name="option-3" label="Option 3" />
  </CheckBoxGroup>
)

export const WithValues = (): React.ReactElement => {
  const [value, setValue] = useState<{
    'option-1': boolean
    'option-2': boolean
    'option-3': boolean
  }>({ 'option-1': true, 'option-2': false, 'option-3': false })
  return (
    <Flex>
      <CheckBoxGroup
        name="controller"
        label="Controller"
        checked={value}
        onChange={({ target }: any) =>
          setValue((existingValue) => ({ ...existingValue, [target.name]: target.checked }))
        }
      >
        <CheckBoxGroup.Item name="option-1" label="Option 1" />
        <CheckBoxGroup.Item name="option-2" label="Option 2" />
        <CheckBoxGroup.Item name="option-3" label="Option 3" />
      </CheckBoxGroup>
      <CheckBoxGroup name="follower" label="Follower">
        <CheckBoxGroup.Item checked={value['option-1']} name="option-4" label="Option 4" />
        <CheckBoxGroup.Item checked={value['option-2']} name="option-5" label="Option 5" />
        <CheckBoxGroup.Item checked={value['option-3']} name="option-6" label="Option 6" />
      </CheckBoxGroup>
    </Flex>
  )
}
