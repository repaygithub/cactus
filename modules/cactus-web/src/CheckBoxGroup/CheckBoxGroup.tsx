import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'
import { MarginProps, WidthProps } from 'styled-system'

import { FieldProps, useAccessibleField } from '../AccessibleField/AccessibleField'
import { TooltipAlignment } from '../AccessibleField/AccessibleField'
import Box from '../Box/Box'
import CheckBoxField, { CheckBoxFieldProps } from '../CheckBoxField/CheckBoxField'
import Fieldset from '../Fieldset/Fieldset'
import { cloneAll } from '../helpers/react'
import Label from '../Label/Label'
import StatusMessage from '../StatusMessage/StatusMessage'
import Tooltip from '../Tooltip/Tooltip'
interface CheckBoxGroupProps
  extends MarginProps,
    WidthProps,
    Omit<FieldProps, 'labelProps'>,
    Omit<React.FieldsetHTMLAttributes<HTMLFieldSetElement>, 'name' | 'defaultValue'> {
  checked?: { [K: string]: boolean }
  required?: boolean
  disableTooltip?: boolean
}

interface LabelWrapper {
  alignTooltip?: TooltipAlignment
}

type CheckBoxGroupItemProps = Omit<CheckBoxFieldProps, 'required'>
const CheckBoxGroupItem = React.forwardRef<HTMLInputElement, CheckBoxGroupItemProps>(
  (props: any, ref) => <CheckBoxField ref={ref} {...props} />
)
CheckBoxGroupItem.displayName = 'CheckBoxGroup.Item'

type ForwardProps = {
  disabled?: boolean
  required?: boolean
}

const noop = () => undefined

export const CheckBoxGroup = React.forwardRef<HTMLFieldSetElement, CheckBoxGroupProps>(
  (
    {
      label,
      children,
      tooltip,
      required,
      checked,
      onFocus,
      onBlur,
      disableTooltip,
      autoTooltip = true,
      alignTooltip = 'right',
      ...props
    },
    ref
  ) => {
    const {
      fieldId,
      ariaDescribedBy,
      labelId,
      statusId,
      name,
      status,
      statusMessage,
      tooltipId,
      disabled,
    } = useAccessibleField({ ...props, tooltip })
    const [showTooltip, setTooltipVisible] = React.useState<boolean>(false)

    const forwardProps: ForwardProps = { required }
    if (disabled === true) {
      forwardProps.disabled = disabled
    }

    const hasOnChange = !!props.onChange
    const cloneWithValue = (element: React.ReactElement, props: any) => {
      if (checked !== undefined) {
        props = { ...props, checked: checked[element.props.name] || false }
      }
      // This is to avert a PropTypes warning regarding missing onChange handler.
      const hasChecked = props.checked !== undefined || element.props.checked !== undefined
      if (hasChecked && hasOnChange && !element.props.onChange) {
        props = { ...props, onChange: noop }
      }
      return React.cloneElement(element, props)
    }
    children = cloneAll(children, forwardProps, cloneWithValue)

    const handleFocus = React.useCallback(
      (event: React.FocusEvent<HTMLFieldSetElement>): void => {
        onFocus?.(event)
        if (!event.currentTarget.contains(event.relatedTarget as Node)) {
          setTooltipVisible(() => true)
        }
      },
      [onFocus]
    )

    const handleBlur = React.useCallback(
      (e: React.FocusEvent<HTMLFieldSetElement>) => {
        onBlur?.(e)
        if (!e.currentTarget.contains(e.relatedTarget as Node)) {
          setTooltipVisible(() => false)
        }
      },
      [onBlur, setTooltipVisible]
    )

    return (
      <Fieldset
        {...props}
        ref={ref}
        id={fieldId}
        aria-describedby={ariaDescribedBy}
        name={name}
        disabled={disabled}
        onFocus={handleFocus}
        onBlur={handleBlur}
      >
        <LabelWrapper id={labelId} as="legend" alignTooltip={alignTooltip}>
          {label}
          {tooltip && (
            <Tooltip
              id={tooltipId}
              label={tooltip}
              disabled={disableTooltip ?? disabled}
              forceVisible={autoTooltip ? showTooltip : false}
            />
          )}
        </LabelWrapper>

        <Box mx={4} pt={3}>
          {children}
        </Box>

        {status && (
          <div>
            <StatusMessage status={status} id={statusId}>
              {statusMessage}
            </StatusMessage>
          </div>
        )}
      </Fieldset>
    )
  }
)

const LabelWrapper = styled(Label)<LabelWrapper>`
  flex-wrap: nowrap;
  display: flex;
  justify-content: ${(p) => (p.alignTooltip === 'right' ? 'space-between' : 'flex-start')};
  padding-right: 8px;
  ${Tooltip} {
    position: relative;
    right: 0;
    bottom: 0;
    padding-left: ${(p) => (p.alignTooltip === 'right' ? 0 : '8px')};
  }
`

CheckBoxGroup.displayName = 'CheckBoxGroup'

CheckBoxGroup.propTypes = {
  id: PropTypes.string,
  name: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  required: PropTypes.bool,
  label: PropTypes.node.isRequired,
  tooltip: PropTypes.node,
  success: PropTypes.node,
  warning: PropTypes.node,
  error: PropTypes.node,
  checked: PropTypes.objectOf(PropTypes.bool.isRequired),
}

type CheckBoxGroupType = typeof CheckBoxGroup & { Item: typeof CheckBoxGroupItem }

const DefaultCheckBoxGroup = CheckBoxGroup as any
DefaultCheckBoxGroup.Item = CheckBoxGroupItem

export default DefaultCheckBoxGroup as CheckBoxGroupType
