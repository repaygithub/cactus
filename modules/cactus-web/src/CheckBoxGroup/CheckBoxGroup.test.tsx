import { render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import * as React from 'react'

import { StyleProvider } from '../StyleProvider/StyleProvider'
import CheckBoxGroup from './CheckBoxGroup'

describe('component: CheckBoxGroup', (): void => {
  test('should render a checkbox group', (): void => {
    const { container, getByText } = render(
      <StyleProvider>
        <CheckBoxGroup
          id="cbg"
          name="checkboxes"
          label="Checkboxes"
          required
          checked={{ cb1: true }}
          tooltip="Check some boxes"
          onChange={() => undefined}
        >
          <CheckBoxGroup.Item id="cb1" name="cb1" label="CB 1" />
          <CheckBoxGroup.Item id="cb2" name="cb2" label="CB 2" />
          <CheckBoxGroup.Item id="cb3" name="cb3" label="CB 3" disabled={true} />
        </CheckBoxGroup>
      </StyleProvider>
    )

    expect(container).toMatchSnapshot()
    const cbs = container.querySelectorAll('input')
    cbs.forEach((cb, ix) => {
      expect(cb.required).toBe(true)
      expect(cb.disabled).toBe(ix === 2)
      expect(cb.checked).toBe(ix === 0)
    })

    const tooltip = getByText('Check some boxes')
    expect(tooltip).toHaveAttribute('id', 'cbg-tip')
    expect(tooltip).toHaveAttribute('role', 'tooltip')
  })

  test('should render a disabled checkbox group', (): void => {
    const { container, getByText } = render(
      <StyleProvider>
        <CheckBoxGroup
          id="cbg"
          name="checkboxes"
          label="Checkboxes"
          disabled
          checked={{ cb1: true }}
          error="No box checking for you"
        >
          <CheckBoxGroup.Item id="cb1" name="cb1" label="CB 1" />
          <CheckBoxGroup.Item id="cb2" name="cb2" label="CB 2" />
          <CheckBoxGroup.Item id="cb3" name="cb3" label="CB 3" disabled={true} />
        </CheckBoxGroup>
      </StyleProvider>
    )

    expect(container).toMatchSnapshot()
    expect(container.querySelector('fieldset')?.disabled).toBe(true)
    const cbs = container.querySelectorAll('input')
    cbs.forEach((cb, ix) => {
      expect(cb.required).toBe(false)
      expect(cb.disabled).toBe(true)
      expect(cb.checked).toBe(ix === 0)
    })

    const err = getByText('No box checking for you').closest('div')
    expect(err).toHaveAttribute('id', 'cbg-status')
    expect(err).toHaveAttribute('role', 'alert')
  })

  test('should trigger events', () => {
    const changes: [string, boolean][] = []
    const onChange = jest.fn((e) => changes.push([e.target.name, e.target.checked]))
    const changeOne: [string, boolean][] = []
    const onChangeOne = jest.fn((e) => changeOne.push([e.target.name, e.target.checked]))
    const focusNames: string[] = []
    const onFocus = jest.fn((e) => focusNames.push(e.target.name))
    const focusOne: string[] = []
    const onFocusOne = jest.fn((e) => focusOne.push(e.target.name))
    const blurNames: string[] = []
    const onBlur = jest.fn((e) => blurNames.push(e.target.name))
    const { getByLabelText } = render(
      <StyleProvider>
        <CheckBoxGroup
          id="cbg"
          name="checkboxes"
          label="Checkboxes"
          onChange={onChange}
          onFocus={onFocus}
          onBlur={onBlur}
        >
          <CheckBoxGroup.Item name="cb1" label="CB 1" onFocus={onFocusOne} />
          <CheckBoxGroup.Item name="cb2" label="CB 2" onChange={onChangeOne} />
          <CheckBoxGroup.Item name="cb3" label="CB 3" disabled />
        </CheckBoxGroup>
      </StyleProvider>
    )

    userEvent.tab()
    expect(onChange).not.toHaveBeenCalled()
    expect(onChangeOne).not.toHaveBeenCalled()
    expect(onFocus).toHaveBeenCalledTimes(1)
    expect(onFocusOne).toHaveBeenCalledTimes(1)
    expect(onBlur).not.toHaveBeenCalled()

    userEvent.click(getByLabelText('CB 3'))
    expect(onChange).not.toHaveBeenCalled()
    expect(onChangeOne).not.toHaveBeenCalled()
    expect(onFocus).toHaveBeenCalledTimes(1)
    expect(onFocusOne).toHaveBeenCalledTimes(1)
    expect(onBlur).not.toHaveBeenCalled()

    userEvent.click(getByLabelText('CB 2'))
    expect(onChange).toHaveBeenCalledTimes(1)
    expect(onChangeOne).toHaveBeenCalledTimes(1)
    expect(onFocus).toHaveBeenCalledTimes(2)
    expect(onFocusOne).toHaveBeenCalledTimes(1)
    expect(onBlur).toHaveBeenCalledTimes(1)

    userEvent.click(getByLabelText('CB 1'))
    expect(onChange).toHaveBeenCalledTimes(2)
    expect(onChangeOne).toHaveBeenCalledTimes(1)
    expect(onFocus).toHaveBeenCalledTimes(3)
    expect(onFocusOne).toHaveBeenCalledTimes(2)
    expect(onBlur).toHaveBeenCalledTimes(2)

    userEvent.tab()
    expect(changes).toEqual([
      ['cb2', true],
      ['cb1', true],
    ])
    expect(changeOne).toEqual([['cb2', true]])
    expect(focusNames).toEqual(['cb1', 'cb2', 'cb1', 'cb2'])
    expect(focusOne).toEqual(['cb1', 'cb1'])
    expect(blurNames).toEqual(['cb1', 'cb2', 'cb1'])
  })
})
