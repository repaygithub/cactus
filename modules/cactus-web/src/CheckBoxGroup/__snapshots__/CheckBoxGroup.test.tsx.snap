// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: CheckBoxGroup should render a checkbox group 1`] = `
.c4 + .c4 {
  margin-top: 16px;
}

.c2 {
  font-size: 18px;
  line-height: 1.5;
  font-weight: 600;
  color: hsl(200,10%,20%);
}

.c15 {
  vertical-align: middle;
}

.c10 {
  vertical-align: middle;
}

.c16 {
  outline: none;
  color: hsl(200,10%,20%);
}

.c16:hover {
  color: hsl(200,96%,35%);
}

.c3 {
  box-sizing: border-box;
  margin-left: 16px;
  margin-right: 16px;
  padding-top: 8px;
}

.c8 {
  opacity: 0;
  border: 0;
  width: 0;
  height: 0;
  position: absolute;
}

.c9 {
  display: inline-block;
  box-sizing: border-box;
  width: 16px;
  height: 16px;
  border: 1px solid;
  border-color: hsl(200,10%,20%);
  background: none;
  border-radius: 1px;
}

.c9 svg {
  margin: 1px;
}

.c9 svg {
  visibility: hidden;
  display: block;
  color: hsl(0,0%,100%);
  width: 12px;
  height: 12px;
}

.c13 {
  display: inline-block;
  box-sizing: border-box;
  width: 16px;
  height: 16px;
  border: 1px solid;
  border-color: hsl(0,0%,90%);
  background: hsl(0,0%,90%);
  border-radius: 1px;
}

.c13 svg {
  margin: 1px;
}

.c13 svg {
  visibility: hidden;
  display: block;
  color: hsl(0,0%,100%);
  width: 12px;
  height: 12px;
}

.c7 {
  position: relative;
  display: inline-block;
  vertical-align: -1px;
  width: 16px;
  height: 16px;
  line-height: 16px;
  cursor: pointer;
}

.c7 input:checked ~ span {
  border-color: hsl(200,96%,35%);
  background-color: hsl(200,96%,35%);
}

.c7 input:checked ~ span svg {
  visibility: visible;
}

.c7 input:focus ~ span {
  box-shadow: 0px 3px 8px hsla(200,96%,35%,0.3);
}

.c12 {
  position: relative;
  display: inline-block;
  vertical-align: -1px;
  width: 16px;
  height: 16px;
  line-height: 16px;
  cursor: cursor;
}

.c12 input:checked ~ span svg {
  visibility: visible;
}

.c12 input:focus ~ span {
  box-shadow: 0px 3px 8px hsla(200,96%,35%,0.3);
}

.c5 + .c5 {
  margin-top: 8px;
}

.c6 .c1 {
  cursor: pointer;
  padding-left: 8px;
}

.c5 + .c5 {
  margin-top: 8px;
}

.c11 .c1 {
  cursor: not-allowed;
  padding-left: 8px;
  color: hsl(0,0%,70%);
}

.c0 {
  position: relative;
  border: 0;
  margin: 0;
  padding: 0;
}

.c0 legend {
  box-sizing: border-box;
  border-bottom: 1px solid currentcolor;
  padding-left: 16px;
  padding-right: 28px;
  width: 100%;
  float: left;
  color: currentcolor;
}

.c0 legend + * {
  clear: both;
}

.c0 .c14 {
  position: absolute;
  right: 8px;
  top: 2px;
  font-size: 16px;
}

.c0 .sc-cbeQSR {
  margin-top: 4px;
}

@media screen and (min-width:1024px) {
  .c2 {
    font-size: 18px;
    line-height: 1.5;
  }
}

<div>
  <fieldset
    aria-describedby="cbg-tip cbg-status"
    class="c0"
    id="cbg"
    name="checkboxes"
  >
    <legend
      class="c1 c2"
      id="cbg-label"
    >
      Checkboxes
    </legend>
    <div
      class="c3"
    >
      <div
        class="c4 c5 c6"
      >
        <label
          class="c7"
          for="cb1"
        >
          <input
            checked=""
            class="c8"
            id="cb1"
            name="cb1"
            required=""
            type="checkbox"
          />
          <span
            aria-hidden="true"
            class="c9"
          >
            <svg
              class="c10"
              fill="currentcolor"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
            >
              <path
                d="M9.81384674,19.7396292 C9.5249491,19.7396292 9.24415138,19.6325301 9.02725315,19.4381317 L3.38789908,14.3504731 C2.906403,13.9166766 2.86770331,13.1741827 3.30239977,12.6926866 C3.73619624,12.2120905 4.47869019,12.1724908 4.96018627,12.6071873 L9.67974783,16.8641526 L18.9127726,5.4369457 C19.3204693,4.93204981 20.0593633,4.85285046 20.5642592,5.26144713 C21.0682551,5.66914381 21.1465544,6.40803779 20.7388578,6.91293368 L10.7273393,19.3040328 C10.524841,19.5542307 10.2269434,19.7108295 9.90654599,19.7369292 C9.87504624,19.7387292 9.84444649,19.7396292 9.81384674,19.7396292"
              />
            </svg>
          </span>
        </label>
        <label
          class="c1 c2"
          for="cb1"
        >
          CB 1
        </label>
      </div>
      <div
        class="c4 c5 c6"
      >
        <label
          class="c7"
          for="cb2"
        >
          <input
            class="c8"
            id="cb2"
            name="cb2"
            required=""
            type="checkbox"
          />
          <span
            aria-hidden="true"
            class="c9"
          >
            <svg
              class="c10"
              fill="currentcolor"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
            >
              <path
                d="M9.81384674,19.7396292 C9.5249491,19.7396292 9.24415138,19.6325301 9.02725315,19.4381317 L3.38789908,14.3504731 C2.906403,13.9166766 2.86770331,13.1741827 3.30239977,12.6926866 C3.73619624,12.2120905 4.47869019,12.1724908 4.96018627,12.6071873 L9.67974783,16.8641526 L18.9127726,5.4369457 C19.3204693,4.93204981 20.0593633,4.85285046 20.5642592,5.26144713 C21.0682551,5.66914381 21.1465544,6.40803779 20.7388578,6.91293368 L10.7273393,19.3040328 C10.524841,19.5542307 10.2269434,19.7108295 9.90654599,19.7369292 C9.87504624,19.7387292 9.84444649,19.7396292 9.81384674,19.7396292"
              />
            </svg>
          </span>
        </label>
        <label
          class="c1 c2"
          for="cb2"
        >
          CB 2
        </label>
      </div>
      <div
        class="c4 c5 c11"
      >
        <label
          class="c12"
          for="cb3"
        >
          <input
            class="c8"
            disabled=""
            id="cb3"
            name="cb3"
            required=""
            type="checkbox"
          />
          <span
            aria-hidden="true"
            class="c13"
            disabled=""
          >
            <svg
              class="c10"
              fill="currentcolor"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
            >
              <path
                d="M9.81384674,19.7396292 C9.5249491,19.7396292 9.24415138,19.6325301 9.02725315,19.4381317 L3.38789908,14.3504731 C2.906403,13.9166766 2.86770331,13.1741827 3.30239977,12.6926866 C3.73619624,12.2120905 4.47869019,12.1724908 4.96018627,12.6071873 L9.67974783,16.8641526 L18.9127726,5.4369457 C19.3204693,4.93204981 20.0593633,4.85285046 20.5642592,5.26144713 C21.0682551,5.66914381 21.1465544,6.40803779 20.7388578,6.91293368 L10.7273393,19.3040328 C10.524841,19.5542307 10.2269434,19.7108295 9.90654599,19.7369292 C9.87504624,19.7387292 9.84444649,19.7396292 9.81384674,19.7396292"
              />
            </svg>
          </span>
        </label>
        <label
          class="c1 c2"
          for="cb3"
        >
          CB 3
        </label>
      </div>
    </div>
    <span
      class="c14 "
      data-reach-tooltip-trigger=""
    >
      <svg
        class="c15 c16"
        fill="currentcolor"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M22,12 C22,6.47714622 17.5223159,2 12,2 C6.47523326,2 2,6.47577095 2,12 C2,17.524229 6.47523326,22 12,22 C17.5223159,22 22,17.5228538 22,12 Z M24,12 C24,18.6274771 18.6268316,24 12,24 C5.37060996,24 0,18.6287448 0,12 C0,5.37125524 5.37060996,0 12,0 C18.6268316,0 24,5.37252293 24,12 Z M11.057,15.6442233 L11.057,12.5 L9.27493729,12.5 C8.72265254,12.5 8.27493729,12.0522847 8.27493729,11.5 C8.27493729,10.9477153 8.72265254,10.5 9.27493729,10.5 L12.057,10.5 C12.6092847,10.5 13.057,10.9477153 13.057,11.5 L13.057,15.6442233 L15.0505322,15.6442233 C15.602817,15.6442233 16.0505322,16.0919386 16.0505322,16.6442233 C16.0505322,17.1965081 15.602817,17.6442233 15.0505322,17.6442233 L9,17.6442233 C8.44771525,17.6442233 8,17.1965081 8,16.6442233 C8,16.0919386 8.44771525,15.6442233 9,15.6442233 L11.057,15.6442233 Z M11.5,9 C10.6715729,9 10,8.32842712 10,7.5 C10,6.67157288 10.6715729,6 11.5,6 C12.3284271,6 13,6.67157288 13,7.5 C13,8.32842712 12.3284271,9 11.5,9 Z"
        />
      </svg>
    </span>
    <span
      id="cbg-tip"
      role="tooltip"
      style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap; word-wrap: normal;"
    >
      Check some boxes
    </span>
  </fieldset>
</div>
`;

exports[`component: CheckBoxGroup should render a disabled checkbox group 1`] = `
.c4 + .c4 {
  margin-top: 16px;
}

.c2 {
  font-size: 18px;
  line-height: 1.5;
  font-weight: 600;
  color: hsl(200,10%,20%);
}

.c15 {
  vertical-align: middle;
}

.c11 {
  vertical-align: middle;
}

.c13 {
  padding: 2px 4px;
  position: relative;
  box-sizing: border-box;
  overflow-wrap: break-word;
  display: inline-block;
  font-size: 15px;
  line-height: 1.6;
  background-color: hsla(353,84%,44%,0.3);
}

.c13 .c14,
.c13 .sc-hOPghE,
.c13 .c10 {
  margin-right: 4px;
  vertical-align: -2px;
}

.c3 {
  box-sizing: border-box;
  margin-left: 16px;
  margin-right: 16px;
  padding-top: 8px;
}

.c8 {
  opacity: 0;
  border: 0;
  width: 0;
  height: 0;
  position: absolute;
}

.c9 {
  display: inline-block;
  box-sizing: border-box;
  width: 16px;
  height: 16px;
  border: 1px solid;
  border-color: hsl(0,0%,90%);
  background: hsl(0,0%,90%);
  border-radius: 1px;
}

.c9 svg {
  margin: 1px;
}

.c9 svg {
  visibility: hidden;
  display: block;
  color: hsl(0,0%,100%);
  width: 12px;
  height: 12px;
}

.c7 {
  position: relative;
  display: inline-block;
  vertical-align: -1px;
  width: 16px;
  height: 16px;
  line-height: 16px;
  cursor: cursor;
}

.c7 input:checked ~ span svg {
  visibility: visible;
}

.c7 input:focus ~ span {
  box-shadow: 0px 3px 8px hsla(200,96%,35%,0.3);
}

.c5 + .c5 {
  margin-top: 8px;
}

.c6 .c1 {
  cursor: not-allowed;
  padding-left: 8px;
  color: hsl(0,0%,70%);
}

.c0 {
  position: relative;
  border: 0;
  margin: 0;
  padding: 0;
}

.c0 legend {
  box-sizing: border-box;
  border-bottom: 1px solid currentcolor;
  padding-left: 16px;
  padding-right: 28px;
  width: 100%;
  float: left;
  color: hsl(0,0%,70%);
}

.c0 legend + * {
  clear: both;
}

.c0 .sc-fWWZrV {
  position: absolute;
  right: 8px;
  top: 2px;
  font-size: 16px;
}

.c0 .c12 {
  margin-top: 4px;
}

@media screen and (min-width:1024px) {
  .c2 {
    font-size: 18px;
    line-height: 1.5;
  }
}

@media screen and (min-width:1024px) {
  .c13 {
    font-size: 15px;
    line-height: 1.6;
  }
}

<div>
  <fieldset
    aria-describedby="cbg-tip cbg-status"
    class="c0"
    disabled=""
    id="cbg"
    name="checkboxes"
  >
    <legend
      class="c1 c2"
      id="cbg-label"
    >
      Checkboxes
    </legend>
    <div
      class="c3"
    >
      <div
        class="c4 c5 c6"
      >
        <label
          class="c7"
          for="cb1"
        >
          <input
            checked=""
            class="c8"
            disabled=""
            id="cb1"
            name="cb1"
            type="checkbox"
          />
          <span
            aria-hidden="true"
            class="c9"
            disabled=""
          >
            <svg
              class="c10 c11"
              fill="currentcolor"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
            >
              <path
                d="M9.81384674,19.7396292 C9.5249491,19.7396292 9.24415138,19.6325301 9.02725315,19.4381317 L3.38789908,14.3504731 C2.906403,13.9166766 2.86770331,13.1741827 3.30239977,12.6926866 C3.73619624,12.2120905 4.47869019,12.1724908 4.96018627,12.6071873 L9.67974783,16.8641526 L18.9127726,5.4369457 C19.3204693,4.93204981 20.0593633,4.85285046 20.5642592,5.26144713 C21.0682551,5.66914381 21.1465544,6.40803779 20.7388578,6.91293368 L10.7273393,19.3040328 C10.524841,19.5542307 10.2269434,19.7108295 9.90654599,19.7369292 C9.87504624,19.7387292 9.84444649,19.7396292 9.81384674,19.7396292"
              />
            </svg>
          </span>
        </label>
        <label
          class="c1 c2"
          for="cb1"
        >
          CB 1
        </label>
      </div>
      <div
        class="c4 c5 c6"
      >
        <label
          class="c7"
          for="cb2"
        >
          <input
            class="c8"
            disabled=""
            id="cb2"
            name="cb2"
            type="checkbox"
          />
          <span
            aria-hidden="true"
            class="c9"
            disabled=""
          >
            <svg
              class="c10 c11"
              fill="currentcolor"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
            >
              <path
                d="M9.81384674,19.7396292 C9.5249491,19.7396292 9.24415138,19.6325301 9.02725315,19.4381317 L3.38789908,14.3504731 C2.906403,13.9166766 2.86770331,13.1741827 3.30239977,12.6926866 C3.73619624,12.2120905 4.47869019,12.1724908 4.96018627,12.6071873 L9.67974783,16.8641526 L18.9127726,5.4369457 C19.3204693,4.93204981 20.0593633,4.85285046 20.5642592,5.26144713 C21.0682551,5.66914381 21.1465544,6.40803779 20.7388578,6.91293368 L10.7273393,19.3040328 C10.524841,19.5542307 10.2269434,19.7108295 9.90654599,19.7369292 C9.87504624,19.7387292 9.84444649,19.7396292 9.81384674,19.7396292"
              />
            </svg>
          </span>
        </label>
        <label
          class="c1 c2"
          for="cb2"
        >
          CB 2
        </label>
      </div>
      <div
        class="c4 c5 c6"
      >
        <label
          class="c7"
          for="cb3"
        >
          <input
            class="c8"
            disabled=""
            id="cb3"
            name="cb3"
            type="checkbox"
          />
          <span
            aria-hidden="true"
            class="c9"
            disabled=""
          >
            <svg
              class="c10 c11"
              fill="currentcolor"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
            >
              <path
                d="M9.81384674,19.7396292 C9.5249491,19.7396292 9.24415138,19.6325301 9.02725315,19.4381317 L3.38789908,14.3504731 C2.906403,13.9166766 2.86770331,13.1741827 3.30239977,12.6926866 C3.73619624,12.2120905 4.47869019,12.1724908 4.96018627,12.6071873 L9.67974783,16.8641526 L18.9127726,5.4369457 C19.3204693,4.93204981 20.0593633,4.85285046 20.5642592,5.26144713 C21.0682551,5.66914381 21.1465544,6.40803779 20.7388578,6.91293368 L10.7273393,19.3040328 C10.524841,19.5542307 10.2269434,19.7108295 9.90654599,19.7369292 C9.87504624,19.7387292 9.84444649,19.7396292 9.81384674,19.7396292"
              />
            </svg>
          </span>
        </label>
        <label
          class="c1 c2"
          for="cb3"
        >
          CB 3
        </label>
      </div>
    </div>
    <div>
      <div
        class="c12 c13"
        id="cbg-status"
        role="alert"
      >
        <svg
          aria-hidden="true"
          class="c14 c15"
          fill="currentcolor"
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            d="M11.1055728,1.5527864 C11.4740971,0.815737865 12.5259029,0.815737865 12.8944272,1.5527864 L22.8944272,21.5527864 C23.2268777,22.2176875 22.743382,23 22,23 L2,23 C1.25661798,23 0.773122263,22.2176875 1.10557281,21.5527864 L11.1055728,1.5527864 Z M3.61803399,21 L20.381966,21 L12,4.23606798 L3.61803399,21 Z M12,18 C11.4477153,18 11,17.5522847 11,17 C11,16.4477153 11.4477153,16 12,16 C12.5522847,16 13,16.4477153 13,17 C13,17.5522847 12.5522847,18 12,18 Z M13,14 C13,14.5522847 12.5522847,15 12,15 C11.4477153,15 11,14.5522847 11,14 L11,11 C11,10.4477153 11.4477153,10 12,10 C12.5522847,10 13,10.4477153 13,11 L13,14 Z"
          />
        </svg>
        <span>
          No box checking for you
        </span>
      </div>
    </div>
  </fieldset>
</div>
`;
